pkg/kube/client.go

```diff
@@ -686,12 +685,27 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 
 	c.Log("Watching pod %s for completion with timeout of %v", info.Name, timeout)
 	_, err = watch.Until(timeout, w, func(e watch.Event) (bool, error) {
-		return conditions.PodCompleted(e)
+		return isPodComplete(e)
 	})
 
 	return err
 }
 
+func isPodComplete(event watch.Event) (bool, error) {
+	o, ok := event.Object.(*core.Pod)
+	if !ok {
+		return true, fmt.Errorf("expected a *core.Pod, got %T", event.Object)
+	}
+	if event.Type == watch.Deleted {
```

Comment:
is it possible to re-call `conditions.PodCompleted` after we've done the type checking? e.g.

```
func isPodComplete(event watch.Event) (bool, error) {
	o, ok := event.Object.(*core.Pod)
	if !ok {
		return true, fmt.Errorf("expected a *core.Pod, got %T", event.Object)
	}
	return conditions.PodCompleted(e)
}
```

From first glance, that seems to be the only difference here between `isPodComplete` and `conditions.PodCompleted`. Is there something else I'm missing? :)

Reply:
gotcha. lgtm then!